Userland/Libraries/LibDiff/Hunks.cpp

```diff
@@ -10,45 +10,100 @@
 
 namespace Diff {
 
-ErrorOr<Vector<Hunk>> parse_hunks(StringView diff)
+Optional<HunkLocation> Parser::consume_unified_location()
 {
-    Vector<StringView> diff_lines = diff.split_view('\n');
-    if (diff_lines.is_empty())
-        return Vector<Hunk> {};
+    auto consume_range = [this](Range& range) {
+        if (!consume_line_number(range.start_line))
+            return false;
 
-    Vector<Hunk> hunks;
+        if (consume_specific(',')) {
+            if (!consume_line_number(range.number_of_lines))
+                return false;
+        } else {
+            range.number_of_lines = 1;
+        }
+        return true;
+    };
 
-    size_t line_index = 0;
-    HunkLocation current_location {};
+    if (!consume_specific("@@ -"))
+        return {};
 
-    // Skip to first hunk
-    while (diff_lines[line_index][0] != '@') {
-        ++line_index;
-    }
+    HunkLocation location;
+
+    if (!consume_range(location.old_range))
+        return {};
+
+    if (!consume_specific(" +"))
+        return {};
+
+    if (!consume_range(location.new_range))
+        return {};
+
+    if (!consume_specific(" @@"))
+        return {};
+
+    return location;
+}
 
-    while (line_index < diff_lines.size()) {
-        if (diff_lines[line_index][0] == '@') {
-            current_location = parse_hunk_location(diff_lines[line_index]);
-            ++line_index;
+bool Parser::consume_line_number(size_t& number)
+{
+    auto line = consume_until([](char ch) { return ch < '0' || ch > '9'; });
```

Comment:
This can be written more simply:
```suggestion
    auto line = consume_while(is_ascii_digit);
```

(You'll need to include `AK/CharacterTypes.h` if it's not already included.)

Reply:
Thanks @AtkinsSJ , that's much nicer!